### PR TITLE
CRUD Word

### PR DESCRIPTION
**Pull request by abdullahallmehedi**

### CODE REVIEW

File: app/helpers/words_helper.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module WordsHelper
```
Comment (chuanhtuan) [inline app/helpers/words_helper.rb]: I think, for better, please remove WordsHelper

File: app/helpers/words_helper.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    @word = Word.find params[:id]
+    @categories = Category.all
+  end
+  
+  def find_words
+    @words = Word.paginate page: params[:page]
+  end
+  
+  def word_params
+    params.require(:word).permit :content, :category_id
```
Comment (chuanhtuan) [inline app/helpers/words_helper.rb]: when you create a word, you have to nested answers

File: app/helpers/words_helper.rb (hunk 1 above)
Comment (abdullahallmehedi) [inline app/helpers/words_helper.rb]: Where should I put those methods ??

File: app/helpers/words_helper.rb (hunk 2 above)
Comment (abdullahallmehedi) [inline app/helpers/words_helper.rb]: This is just CRUD Word task. I will re factor it when do CRUD WordAnswer task (44232).

File: app/helpers/words_helper.rb (hunk 1 above)
Comment (chuanhtuan) [inline app/helpers/words_helper.rb]: put it to controller

File: app/controllers/admin/words_controller.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
+class Admin::WordsController < ApplicationController
+  before_action :logged_in_user
+  before_action :find_words, only: :index
```
Comment (chuanhtuan) [inline app/controllers/admin/words_controller.rb]: should merge 3 method to 1
find_words, find_word, get_categories => load_objects

Comment (chuanhtuan) [inline app/controllers/admin/words_controller.rb]: in create action, you only load_objects when save is failed

File: app/controllers/words_controller.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class WordsController < ApplicationController
+  before_action :logged_in_user
+  before_action :find_words
```
Comment (chuanhtuan) [inline app/controllers/words_controller.rb]: remove

File: app/controllers/words_controller.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+   
+  def index
+  end
+  
+  private
+  def find_words
+    @words = Word.paginate page: params[:page]
```
Comment (chuanhtuan) [inline app/controllers/words_controller.rb]: write in index method

File: app/views/admin/words/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+	  
+      <%= f.label :content %>
+      <%= f.text_field :content %>
+	  
+      <%= f.label :category %>  
+      <%= f.collection_select :category_id, @categories, :id, :name, {prompt: t(:select_category)} %>
+	  
+
+      <%= f.fields_for :word_answers do |word_answer_form| %>
+        <%= render "word_answer_fields", :f => word_answer_form %>
```
Comment (chuanhtuan) [inline app/views/admin/words/edit.html.erb]: <%= render "word_answer_fields", f: word_answer_form %>

File: app/views/admin/words/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+      <%= render 'shared/error', object: f.object %>
+	  
+      <%= f.label :content %>
+      <%= f.text_field :content %>
+	  
+      <%= f.label :category %>  
+      <%= f.collection_select :category_id, @categories, :id, :name, {prompt: t(:select_category)} %>
+	  
+      <%= f.fields_for :word_answers do |builder| %>
+        <%= render "word_answer_fields", :f => builder %>
```
Comment (chuanhtuan) [inline app/views/admin/words/new.html.erb]: <%= render "word_answer_fields", f: builder %>

File: app/controllers/admin/words_controller.rb [hunk 8]
```diff
@@ -0,0 +1,51 @@
+class Admin::WordsController < ApplicationController
+  before_action :logged_in_user
+  before_action :load_objects, except: [:index]
```
Comment (chuanhtuan) [inline app/controllers/admin/words_controller.rb]: in create method, only load_objects when save failed

File: app/views/words/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,16 @@
+<%= will_paginate %>
+
+<% @words.each do |word| %>
+<% cycle :odd, :even %>
+<%= raw '<div class="row">' if current_cycle == 'odd' %>
+  <div class="col-xs-2">
+  	<%= word.content %>
+  </div>
+  <div class="col-xs-3 <%= 'divider-vertical' if current_cycle == 'odd' %>">	  	
+  	<%= word.word_answers.where(correct: true).map{|word_answer| word_answer.content}.join(", ") %>
```
Comment (chuanhtuan) [inline app/views/words/index.html.erb]: any where clause have to write in a scope

File: app/views/words/index.html.erb [hunk 10]
```diff
@@ -0,0 +1,16 @@
+<%= will_paginate %>
+
+<% @words.each do |word| %>
+<% cycle :odd, :even %>
+<%= raw '<div class="row">' if current_cycle == 'odd' %>
+  <div class="col-xs-2">
+  	<%= word.content %>
+  </div>
+  <div class="col-xs-3 <%= 'divider-vertical' if current_cycle == 'odd' %>">	  	
+  	<%= word.word_answers.map{|word_answer| word_answer.content if word_answer.correct?}.compact.join(", ") %>
```
Comment (chuanhtuan) [inline app/views/words/index.html.erb]: only 1 answer is correct for each word
and you should use a scope named: correct

**Pull request merged**